portlets/src/main/webapp/vue-app/engagement-center/components/EngagementCenter.vue

```diff
@@ -57,8 +67,39 @@ export default {
   },
   data: () => ({
     engagementCenterEnabled: eXo.env.portal.engagementCenterEnabled,
+    loading: true,
     tab: null,
     earnerId: eXo.env.portal.userIdentityId,
   }),
+  watch: {
+    tab() {
+      if (this.tab === 0) {
+        window.history.pushState('gamificationCenter', 'My programs', `${eXo.env.portal.context}/${eXo.env.portal.portalName}/${eXo.env.portal.global.engagement}/programs`);
+      } else if (this.tab === 1) {
+        window.history.pushState('gamificationCenter', 'My challenges', `${eXo.env.portal.context}/${eXo.env.portal.portalName}/${eXo.env.portal.global.engagement}/challenges`);
+      } else if (this.tab === 2) {
+        window.history.pushState('gamificationCenter', 'My achievements', `${eXo.env.portal.context}/${eXo.env.portal.portalName}/${eXo.env.portal.global.engagement}/achievements`);
+      }
+    },
+    loading() {
+      if (!this.loading) {
+        const urlPath = document.location.search || document.location.pathname;
+        if (urlPath === `${eXo.env.portal.context}/${eXo.env.portal.portalName}/${eXo.env.portal.global.engagement}/programs`) {
+          this.tab = 0;
+        } else if (urlPath === `${eXo.env.portal.context}/${eXo.env.portal.portalName}/${eXo.env.portal.global.engagement}/challenges`) {
+          this.tab = 1;
+        } else if  (urlPath === `${eXo.env.portal.context}/${eXo.env.portal.portalName}/${eXo.env.portal.global.engagement}/achievements`) {
+          this.tab = 2;
+        }
+        else {
+          this.tab = 0;
+          window.history.pushState('gamificationCenter', 'My programs', `${eXo.env.portal.context}/${eXo.env.portal.portalName}/${eXo.env.portal.global.engagement}/programs`);
+        }
```

Comment:
IMO it's better to move the watch loading to mounted and update **initialized** to true after setting the tab value

Reply:
done !